### PR TITLE
Patch/separate deprecated modules

### PR DESCRIPTION
**Pull request by nanoant**
This adds some clear separation for deprecated modules, hopefully this will reduce confusion. Also all currently deprecated modules are correctly marked in `doc/lib.txt`.

Let me know if I missed some module though.

### CODE REVIEW

File: doc/lib.txt [hunk 1]
```diff
@@ -263,11 +254,6 @@ Internet Protocols and Support
 Parsers
 -------
 
-* `parseopt <parseopt.html>`_
```
Comment (Araq) [inline doc/lib.txt]: parseopt is not deprecated anymore.

File: lib/deprecated/pure/parseopt.nim [hunk 2]
```diff
@@ -7,6 +7,12 @@
 #    distribution, for details about the copyright.
 #
 
+## **Note**: This module is deprecated since version 0.9.3.
```
Comment (Araq) [inline lib/deprecated/pure/parseopt.nim]: Nope, parseopt is here to stay. We need a posix compatible option parser in the stdlib at one day though.

Comment (nanoant) [inline lib/deprecated/pure/parseopt.nim]: So both `parseopt` and `parseopt2` are to stay? Shouldn't be there any info what's the purpose of these both?

File: doc/lib.txt [hunk 3]
```diff
@@ -397,6 +383,32 @@ Modules for JS backend
   Declaration of the Document Object Model for the JS backend.
 
 
+Deprecated modules
+------------------
+
+* `asyncio <asyncio.html>`_
+  This module implements an asynchronous event loop for sockets.
+  **Deprecated since version 0.11.2:**
+  Use the `asyncnet <asyncnet.html>`_ together the
```
Comment (dom96) [inline doc/lib.txt]: together with

**Comment by nanoant** [comment]
@Araq removed parseopt from deprecated modules
@dom96 Fixed missing *together*.

**Pull request merged**